sdk/keyvault/azure-security-keyvault-secrets/inc/azure/keyvault/secrets/secret_client.hpp

```diff
@@ -1,14 +1,137 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
+/**
+ * @file
+ * @brief Defines the Key Vault Secret client.
+ *
+ */
 
 #pragma once
+#include "../src/private/keyvault_protocol.hpp"
+#include "dll_import_export.hpp"
 
+#include <azure/keyvault/secrets/keyvault_secret.hpp>
 #include <string>
+
 namespace Azure { namespace Security { namespace KeyVault { namespace Secrets {
 
-  class SecretClient final {
+  class ServiceVersion final {
+  private:
+    std::string m_version;
+
+  public:
+    /**
+     * @brief Construct a new Service Version object
+     *
+     * @param version The string version for the Key Vault keys service.
+     */
+    ServiceVersion(std::string version) : m_version(std::move(version)) {}
+
+    /**
+     * @brief Enable comparing the ext enum.
+     *
+     * @param other Another #ServiceVersion to be compared.
+     */
+    bool operator==(ServiceVersion const& other) const { return m_version == other.m_version; }
+
+    /**
+     * @brief Return the #ServiceVersion string representation.
+     *
+     */
+    std::string const& ToString() const { return m_version; }
+
+    /**
+     * @brief Use to send request to the 7.2 version of Key Vault service.
+     *
+     */
+    AZURE_SECURITY_KEYVAULT_SECRETS_DLLEXPORT static const ServiceVersion V7_2;
+  };
+
+  /**
+   * @brief Define the options to create an SDK Keys client.
+   *
+   */
+  struct SecretClientOptions final : public Azure::Core::_internal::ClientOptions
+  {
+    ServiceVersion Version;
+
+    /**
+     * @brief Construct a new Secret Client Options object.
+     *
+     * @param version Optional version for the client.
+     */
+    SecretClientOptions(ServiceVersion version = ServiceVersion::V7_2)
+        : Azure::Core::_internal::ClientOptions(), Version(version)
+    {
+    }
+  };
+
+  /**
+   * @brief Optional parameters for SecretClient::GetSecret
+   *
+   */
+  struct GetSecretOptions final
```

Comment:
Why are you defining this just for the `version` param that you could default to an empty `std::string`? This would be more consistent with most other languages.

Reply:
similar to the `return Response<T>` guideline, there is this other guideline to group all optional input parameters in a `operationOptions` model type which can grow in the future.  Even though I like and prefer having the version as input, just like Heath is recommending, I would ask @JeffreyRichter.

Reply:
Okay. If this is an idiomatic different and consistent, great.

Reply:
i used the same model as in the keys client , since that one shipped , customers would expect similar behavior